src/thunderbots/software/ai/navigator/BUILD

```diff
@@ -0,0 +1,7 @@
+package(default_visibility = ["//visibility:public"])
+
+cc_library(
+    name = "navigator",
+    hdrs = ["navigator.h"],
+    deps = [],
```

Comment:
`navigator.h` includes `intent`, `primitive` and `world`. These should be dependencies

Reply:
Added all three as dependencies